src/trainer.py

```diff
@@ -1204,22 +1164,16 @@ def _restore_checkpoint(self, phase, tasks=None):
             self._task_infos[task_name]["total_batches_trained"] = task_state[
                 "total_batches_trained"
             ]
-            if "optimizer" in task_state:
-                self._task_infos[task_name]["optimizer"].load_state_dict(task_state["optimizer"])
-                for param, val in task_state["scheduler"].items():
-                    setattr(self._task_infos[task_name]["scheduler"], param, val)
             self._task_infos[task_name]["stopped"] = task_state["stopped"]
             generator = self._task_infos[task_name]["tr_generator"]
             for _ in itertools.islice(
                 generator,
                 task_state["total_batches_trained"] % self._task_infos[task_name]["n_tr_batches"],
             ):
                 pass
-        if task_states["global"]["optimizer"] is not None:
-            self._g_optimizer.load_state_dict(task_states["global"]["optimizer"])
-        if task_states["global"]["scheduler"] is not None:
-            for param, val in task_states["global"]["scheduler"].items():
-                setattr(self._g_scheduler, param, val)
+        self._optimizer.load_state_dict(task_states["global"]["optimizer"])
+        # NOTE(Alex): AllenNLP wrapper doesn't expose scheduler state dict methods
```

Comment:
?

Reply:
Just to explain why underneath we have to do `self._scheduler.lr_scheduler`